virtual_ecosystem/core/base_model.py

```diff
@@ -285,70 +289,47 @@ def _check_model_name(cls, model_name: str) -> str:
         return model_name
 
     @classmethod
-    def _check_required_init_vars(
-        cls, required_init_vars: tuple[str, ...]
+    def _check_model_var_attribute(
+        cls,
+        model_var_attribute_name: str,
+        model_var_attribute_contents: tuple[str, ...],
     ) -> tuple[str, ...]:
-        """Check the required_init_vars property is valid.
+        """Check a model variable attribute property is valid.
+
+        Creating an instance of the BaseModel class requires that several variable
+        attributes are set. Each of these provides a list of variable names that are
+        required or updated by the model at various points. This method is used to
+        validate the structure of the new instance and ensure the resulting model
+        structure is consistent.
 
-        TODO: Remove. Validation is performed at runtime in the variables side.
+        TODO: Diego - this is possibly paranoid. It no longer checks anything about the
+            variables (which is the job of the variables system) but is basically a
+            check on creating the model correctly.
 
```

Comment:
A little bit, but it's harmless and is only run once per model, so totally fine. 

```suggestion
```